systemvm/debian/opt/cloud/bin/setup/cloud-early-config

```diff
@@ -64,10 +71,17 @@ config_guest() {
 
 get_boot_params() {
   case $HYPERVISOR in
-     xen-domU|xen-hvm)
+     xen-pv)
           cat /proc/cmdline > $CMDLINE
           sed -i "s/%/ /g" $CMDLINE
           ;;
+     xen-hvm)
+          if [ ! -f /usr/sbin/xenstore-read ]; then
+            log_it "ERROR: xentools not installed, cannot found xenstore-read" && exit 5
+          fi
+          /usr/sbin/xenstore-read vm-data/cloudstack/init > /var/cache/cloud/cmdline
```

Comment:
@khos2ow in 4.11 systemvmtemplate, the xenstore-read is at /usr/bin, can you check/comment why /usr/sbin was used? I've however, fixed this blocker in the stabilization PR: https://github.com/shapeblue/cloudstack/commit/3dbb2e6b4b98cba9d59b241bd0a8c2a051131cbc

Reply:
That's strange, I really don't know what's going on. I just checked dozens of random VMs, on various Xenserver version (7.x.x+ though) and all of them had `xenstore-read` in both places! I guess using `/usr/bin` would be safer. or even check both places then log an error?
BTW I wrote a comment on your PR as well.

Reply:
@khos2ow I think you're missing a big point and I'm wondering how you're testing your changes. This script runs inside CloudStack's systemvmtemplate that is based on Debian 9 for 4.11+. For the change in 4.11 branch, the test/changes must be performed on Debian9 based guest vm only. In our built template, the path to `xenstore-read` is at `/usr/bin`. Regarding the `virt-what` comment, again the version of virt-what is specific to the Debian 9 based systemvmtemplate that would return xen-hvm in my tests across XenServer 6.5-7.x versions. I'll therefore keep my changes that are tested against 4.11 systemvmtemplate. Temporarily hosted here are latest 4.11 templates if you want to try them: https://lab.yadav.cloud/systemvmtemplates/4.11/

Reply:
Thanks @rhtyd for the update, that's exactly how I have been testing, I'm gonna double check on my side once again, but if you've done the testing and validated the changes are good that's really good news.